src/script/sighashtype.h

```diff
@@ -6,19 +6,11 @@
 #define BITCOIN_SCRIPT_HASH_TYPE_H
 
 #include "serialize.h"
+#include "script/interpreter.h"
 
 #include <cstdint>
 #include <stdexcept>
 
-/** Signature hash types/flags */
-enum {
-    SIGHASH_ALL = 1,
-    SIGHASH_NONE = 2,
-    SIGHASH_SINGLE = 3,
-    SIGHASH_FORKID = 0x40,
-    SIGHASH_ANYONECANPAY = 0x80,
-};
-
 /**
```

Comment:
was there a problem in keeping them in sighashtype.h?  I think that it would make more sense to have removed them from  interpreter.h instead.

Reply:
To be honest, the chain split bug in ABC spooked me a bit and kept me from reworking everything to use `SigHashType` just yet. As I explained in my comments above, I did similarly for the signature encoding checks.

So far, `SigHashType` is *only* used for the test code that has been ported from ABC.

But I have no strong opinion on this and can of course rework this (now or later).